### PR TITLE
Improve building of docs using Py3  [ci skip]

### PR DESCRIPTION
**Pull request by mwichmann**
The doc preparation process works with Py3 these changes. The `Dump` examples do come out different, apparently because of changes in the pretty printer module (`pprint`) in Python's stdlib itself, and the xml signature comes out different, but the content appears the same otherwise.

* context managers on file r/w + use `shutil.copy` where it makes sense.
* lxml wants (demands?) that xml files be processed as bytes
* for the phase where we gen the entity files, read as text anyway
* Need to solve a problem where the generated xml is putting the \n in  literally, not evaluating it.
* Fix some examples broken for py3 - mostly now use `print()` function instead of old statement
* Fix more octal constant instances
* Cleanups suggested by PyCharm: staticmethods, two blanks   before class definition, others.

This addresses issues called out in #3300, but is not a complete solution because the actual doc build step still fails with the epydoc failures (which aren't directly because of Py3; epydoc build doesn't work any better on my system with Py3, even with the forked version with patches).

*Functional* changes are in the bin directory. A funch of xml files in src were updated for Py3 syntax, and that lxml does not allow encoding specification (see [lxml FAQ](https://lxml.de/FAQ.html#why-can-t-lxml-parse-my-xml-from-unicode-strings) on this). Some tests were updated (octal constant syntax).  

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [X] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: bin/SConsDoc.py [hunk 1]
```diff
@@ -817,11 +865,10 @@ def importfile(path):
 else:  # PY3 version, from newer pydoc
     def importfile(path):
         """Import a Python source file or compiled file given its path."""
-        import importlib
-        from pydoc import ErrorDuringImport
-        magic = importlib.util.MAGIC_NUMBER
+        from importlib import MAGIC_NUMBER
```
Comment (bdbaddog) [inline bin/SConsDoc.py]: should that be importlib.util and not importlib?

File: bin/SConsExamples.py [hunk 2]
```diff
@@ -735,8 +735,8 @@ def command_edit(args, c, test, values):
     for file in args:
         if not os.path.isabs(file):
             file = os.path.join(test.workpath('WORK'), file)
-        contents = open(file, 'rb').read()
```
Comment (bdbaddog) [inline bin/SConsExamples.py:738]: did you drop including contents?

File: bin/SConsDoc.py (hunk 1 above)
Comment (mwichmann) [reply bin/SConsDoc.py]: yes

File: bin/SConsExamples.py (hunk 2 above)
Comment (mwichmann) [reply bin/SConsExamples.py:738]: it appends now instead of rewriting

Comment (bdbaddog) [reply bin/SConsExamples.py:738]: Ahh.. gotcha.

**Pull request merged**